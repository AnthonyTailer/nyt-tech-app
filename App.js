import React from 'react'
import { ThemeProvider } from 'react-native-elements'
import { NavigationContainer } from '@react-navigation/native'
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
import { SafeAreaProvider } from 'react-native-safe-area-context'
import NewsList from './src/components/NewsList'
import IconWithBadge from './src/components/IconWithBadge'

const Tab = createBottomTabNavigator()

export default function App() {
  return (
    <SafeAreaProvider>
      <ThemeProvider>
        <NavigationContainer>
          <Tab.Navigator
            initialRouteName="Technology"
            screenOptions={({ route }) => ({
                tabBarIcon: ({ focused, color, size }) => {
                  let iconName

                  if (route.name === 'Technology') {
                    iconName = 'globe'
                  } else if (route.name === 'Science') {
                    iconName = 'chemistry'
                  }

                  // You can return any component that you like here!
                  return <IconWithBadge name={iconName} size={size} color={color} />;
                },
            })}
          >
            <Tab.Screen name="Technology" component={NewsList} initialParams={{ section: 'technology' }} />
            <Tab.Screen name="Science" component={NewsList} initialParams={{ section: 'science' }}/>
          </Tab.Navigator>
        </NavigationContainer>
      </ThemeProvider>
    </SafeAreaProvider>
  );
}
