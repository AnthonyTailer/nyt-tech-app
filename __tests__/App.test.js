import React from 'react'
import { render } from 'react-native-testing-library'

import App from  '../App'

describe('<App/>', () => {
  it('should render correctly', () => {
    const tree = render(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})