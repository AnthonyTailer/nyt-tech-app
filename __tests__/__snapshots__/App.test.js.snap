// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> has 1 child 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`<App/> should render correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
